tests/unit/post_install/test_log_vc_info.py

```diff
@@ -0,0 +1,220 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from pathlib import Path
+import pytest
+from pytest import MonkeyPatch, TempPathFactory
+import shutil
+import subprocess
+from typing import Any, Tuple
+from unittest.mock import Mock
+
+from cylc.flow.post_install.log_vc_info import (
+    get_diff, _get_git_commit, get_status, get_vc_info, main
+)
+
+Fixture = Any
+
+
+BASIC_FLOW_1 = """
+[scheduling]
+    [[graph]]
+        R1 = foo
+"""
+
+BASIC_FLOW_2 = """
+[scheduling]
+    [[graph]]
+        R1 = bar
+"""
+
+
+def skip_if_not_installed(command: str) -> None:
+    """Skip test if command is not installed"""
+    if shutil.which(command) is None:
+        pytest.skip(f"{command} is not installed")
+
+
+@pytest.fixture(scope='module')
+def git_source_repo(tmp_path_factory: TempPathFactory) -> Tuple[str, str]:
+    """Init a git repo for a workflow source dir.
+
+    The repo has a flow.cylc file with uncommitted changes. This dir is reused
+    by all tests requesting it in this module.
+
+    Returns (source_dir_path, commit_hash)
+    """
+    skip_if_not_installed('git')
```

Comment:
Can use `pytest.skipif` as a decorator on the test function to avoid having to run tests that would skip.